src/test/java/com/google/api/generator/engine/ast/TernaryExprTest.java

```diff
@@ -67,6 +64,27 @@ public void validTernaryExpr_objectType() {
     assertEquals(ternaryExpr.type(), TypeNode.STRING);
   }
 
+  @Test
+  public void validTernaryExpr_boxedAndPrimitiveType() {
+    Variable conditionVariable =
+        Variable.builder().setName("condition").setType(TypeNode.BOOLEAN).build();
+    VariableExpr conditionExpr = VariableExpr.builder().setVariable(conditionVariable).build();
+    Variable intVariable = Variable.builder().setName("intValue").setType(TypeNode.INT).build();
+    VariableExpr thenExpr = VariableExpr.builder().setVariable(intVariable).build();
+    Variable integerVariable =
+        Variable.builder().setName("integerValue").setType(TypeNode.INT_OBJECT).build();
+    VariableExpr elseExpr = VariableExpr.builder().setVariable(integerVariable).build();
+    TernaryExpr ternaryExpr =
+        TernaryExpr.builder()
+            .setConditionExpr(conditionExpr)
+            .setThenExpr(thenExpr)
+            .setElseExpr(elseExpr)
+            .build();
+    assertEquals(ternaryExpr.conditionExpr().type(), TypeNode.BOOLEAN);
+    assertEquals(ternaryExpr.thenExpr().type(), ternaryExpr.elseExpr().type());
+    assertEquals(ternaryExpr.type(), TypeNode.INT_OBJECT);
+  }
+
```

Comment:
Could you please confirm that we've covered all the edge cases in these tests?

Reply:
sure, added two valid unit tests that ensure:  the ternary expression has the correct type with primitive/boxed type expressions in different order. 
Also an invalid test to confirm the thenExpr/elseExpr should be primitive/boxed type equal.